### PR TITLE
Sorting out blob and data naming

### PR DESCRIPTION
**Pull request by Affie**

### CODE REVIEW

File: src/DataBlobs/services/BlobEntry.jl [hunk 1]
```diff
@@ -46,3 +65,284 @@ end
...
+end
+
+function getBlobEntry(var::AbstractDFGVariable, blobId::UUID)
+    for (k,v) in var.dataDict
+        if v.id == blobId
+            return v
+        end
+    end
+    throw(
+        KeyError("No dataEntry with blobId $(blobId) found in variable $(getLabel(var))")
```
Comment (dehann) [inline src/DataBlobs/services/BlobEntry.jl:91]:
```suggestion
        KeyError("No blobEntry with blobId $(blobId) found in variable $(getLabel(var))")
```

File: src/DataBlobs/services/BlobEntry.jl [hunk 2]
```diff
@@ -46,3 +65,284 @@ end
...
+
+"""
+    $(SIGNATURES)
+Add Data Entry to a DFG variable
+Should be extended if DFG variable is not returned by reference.
+
+Also see: [`getBlobEntry`](@ref), [`addBlob`](@ref), [`mergeBlobEntries!`](@ref)
+"""
+function addBlobEntry!(var::AbstractDFGVariable, bde::BlobEntry)
+    haskey(var.dataDict, bde.label) && error("Data entry $(bde.label) already exists in variable $(getLabel(var))")
```
Comment (dehann) [inline src/DataBlobs/services/BlobEntry.jl:123]:
```suggestion
    haskey(var.dataDict, bde.label) && error("blobEntry $(bde.label) already exists on variable $(getLabel(var))")
```

File: src/DataBlobs/services/BlobEntry.jl [hunk 3]
```diff
@@ -46,3 +65,284 @@ end
...
+
+
+##==============================================================================
+## BlobEntry - Helper functions, Lists, etc
+##==============================================================================
+
+"""
+    $SIGNATURES
+
+Does a data entry (element) exist at `key`.
```
Comment (dehann) [inline src/DataBlobs/services/BlobEntry.jl:182]:
```suggestion
Does a blob entry (element) exist at `key`.
```

File: src/DataBlobs/services/BlobEntry.jl [hunk 4]
```diff
@@ -46,3 +65,284 @@ end
...
+"""
+hasBlobEntry(var::AbstractDFGVariable, key::Symbol) = haskey(var.dataDict, key)
```
Comment (dehann) [inline src/DataBlobs/services/BlobEntry.jl:184]:
```suggestion
hasBlobEntry(var::AbstractDFGVariable, blobLabel::Symbol) = haskey(var.dataDict, blobLabel)
```

File: src/DataBlobs/services/BlobEntry.jl [hunk 5]
```diff
@@ -46,3 +65,284 @@ end
...
+end
+
+function listBlobEntries(dfg::AbstractDFG, label::Symbol)
+    # !isVariable(dfg, label) && return nothing
+    listBlobEntries(getVariable(dfg, label))
+end
+
+"""
+    $SIGNATURES
+List a collection of data entries per variable that match a particular `pattern::Regex`.
```
Comment (dehann) [inline src/DataBlobs/services/BlobEntry.jl:216]:
```suggestion
List a collection of blob entries per variable that match a particular `pattern::Regex`.
```

File: src/DataBlobs/services/BlobEntry.jl [hunk 6]
```diff
@@ -46,3 +65,284 @@ end
...
+    ents_ = listBlobEntries(dfg, lb)
+    entReg = map(l->match(pattern, string(l)), ents_)
+    entMsk = entReg .!== nothing
+    ents_[findall(entMsk)] |> _sort
+end
+
+"""
+    $SIGNATURES
+
+Add a data entry into the destination variable which already exists
```
Comment (dehann) [inline src/DataBlobs/services/BlobEntry.jl:247]:
```suggestion
Add a blob entry into the destination variable which already exists 
```

File: src/DataBlobs/services/HelpersDataWrapEntryBlob.jl [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+Get the data entry and blob for the specified blobstore or dfg retured as a tuple.
+Related
+[`getBlobEntry`](@ref)
+
+$(METHODLIST)
+"""
+function getData end
+
+"""
+Add a data Entry and Blob to a distributed factor graph or BlobStore.
```
Comment (dehann) [inline src/DataBlobs/services/HelpersDataWrapEntryBlob.jl:18]:
```suggestion
Add both a BlobEntry and Blob to a distributed factor graph or BlobStore.
```

File: src/DataBlobs/services/HelpersDataWrapEntryBlob.jl [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+Related
+[`addBlobEntry!`](@ref)
+
+$(METHODLIST)
+"""
+function addData! end
+
+"""
+Update a data entry or blob to the blob store or dfg.
```
Comment (dehann) [inline src/DataBlobs/services/HelpersDataWrapEntryBlob.jl:27]:
```suggestion
Update a blob entry or blob to the blob store or dfg.
```

File: src/DataBlobs/services/HelpersDataWrapEntryBlob.jl [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+Related
+[`updateBlobEntry!`](@ref)
+
+$(METHODLIST)
+"""
+function updateData! end
+
+"""
+Delete a data entry and blob from the blob store or dfg.
```
Comment (dehann) [inline src/DataBlobs/services/HelpersDataWrapEntryBlob.jl:36]:
```suggestion
Delete a blob entry and blob from the blob store or dfg.
```

File: src/DataBlobs/services/HelpersDataWrapEntryBlob.jl [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+    )
+
+function addData!(
+    dfg::AbstractDFG, 
+    blobstore::AbstractBlobStore, 
+    label::Symbol, 
+    key::Symbol,
+    blob::Vector{UInt8}, 
+    timestamp=now(localzone()); 
+    description="", mimeType = "application/octet-stream",
```
Comment (dehann) [inline src/DataBlobs/services/HelpersDataWrapEntryBlob.jl:141]:
```suggestion
    description="", 
    mimeType = "application/octet-stream", 
```

File: src/DataBlobs/services/BlobEntry.jl [hunk 11]
```diff
@@ -13,11 +36,7 @@ getTimestamp(entry::BlobEntry) = entry.timestamp
 
 
 
-function assertHash(
-    de::BlobEntry, 
-    db::AbstractVector{UInt8}; 
-    hashfunction::Function = sha256
-)
+function assertHash(de::AbstractDataEntry, db; hashfunction::Function = sha256)
```
Comment (dehann) [inline src/DataBlobs/services/BlobEntry.jl:39]:
```suggestion
function assertHash(de::BlobEntry, db; hashfunction::Function = sha256)
```

**Comment by Affie** [comment]
Thanks. I was just about to push fixes but struggling with wifi. I ran out of time. Still have to make sure exports are correct and the high level `data` helpers

**Comment by Affie** [comment]
There are also tests that got broken.

**Comment by dehann** [comment]
thanks Johan, i fixed the data test stuff and exports

File: NEWS.md [hunk 12]
```diff
@@ -1,5 +1,12 @@
 Listing news on any major breaking changes in DFG.  For regular changes, see integrated Github.com project milestones for DFG.
 
+# v0.20
+
+- Close long running serialization redo (#590) using only JSON3.jl and StructTypes.jl going forward.
+- Standardize BlobEntry=>Blob naming of functions, and keeping convenience wrappers `{get,add,update,delete}Data[!]`.
+- Include type field `VariableNodeData.covar`.
```
Comment (dehann) [inline NEWS.md:7]:
```suggestion
- Consolidate to only one `BlobEntry` definition, dropping use of `AbstractBlobEntry`.
- Include type field `VariableNodeData.covar`.
```

**Comment by Affie** [comment]
Ok. I just got a bit of signal and then saw branch has conflicts. I’m sure you got everything so will leave my local changes and just help review here.

**Comment by Affie** [comment]
Haha. Too late again. Aggh slow internet. We can fix in patch if i find something.

**Pull request merged**